Tests/GraphIR/example.gir

```diff
@@ -0,0 +1,29 @@
+module example where
+
+-- data example.Nat : Type where
+--   example.Nat.z : example.Nat
+--   example.Nat.s : example.Nat -> example.Nat
+
+-- data example.Bool : Type where
+--   example.Bool.true example.Bool.false : example.Bool
+
+-- example.equal : example.Nat -> example.Nat -> example.Bool
+-- example.equal example.Nat.z example.Nat.z = example.Bool.true
+-- example.equal example.Nat.z _ = example.Bool.false
+-- example.equal _ example.Nat.z = example.Bool.false
+-- example.equal (example.Nat.s n) (example.Nat.s m) = example.equal n m
+
+@example : (example.Nat) -> (example.Nat -> _) {
+bb0(%0 : example.Nat, %return: (example.Nat) -> _):
+  %1 = function_ref @example.equal
+  %2 = data_init_simple example.Nat.z : example.Nat
+  apply %1(%0, %2, %bb1) : (example.Bool)
+bb1(%3 : example.Bool):
+  cond_br %cond : example.Bool, %bb2, %bb3
+bb2:
```

Comment:
Unlike Thorin, I want GIR basic blocks to take all the parameters they use as basic block arguments.

Reply:
When scheduled, we have explicit Scopes around that make that unnecessary.  The internal representation should continue to do this.

Also, these aren’t basic blocks, they’re just offensively-named continuations.